### PR TITLE
[VarDumper] Fix interval caster with microseconds

### PR DESCRIPTION
**Pull request by maidmaid**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | /
| License       | MIT
| Doc PR        | /

```php
dump((new DateTime('00:00:00'))->diff(new DateTime('00:00:00.123')));

// before
DateInterval {
  interval: + 00:00:00.0.123
}

// after
DateInterval {
  interval: + 00:00:00.123
}
```

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Tests/Caster/DateCasterTest.php [hunk 1]
```diff
@@ -173,34 +170,37 @@ public function testCastInterval($intervalSpec, $invert, $xInterval, $xSeconds)
...
-            array('P4D', 1, '- 4d', '-345 600s'),
-            array('P5M', 1, '- 5m', null),
-            array('P6Y', 1, '- 6y', null),
-            array('P1Y2M3DT4H5M6S', 1, '- 1y 2m 3d 04:05:06'.$ms, null),
-            array('PT1M60S', 1, '- 00:02:00'.$ms, null),
-            array('PT1H60M', 1, '- 02:00:00'.$ms, null),
-            array('P1DT24H', 1, '- 2d', null),
-            array('P1M32D', 1, '- 1m 32d', null),
+            array('PT0S', 0, 0, '0s', '0s'),
+            array('PT0S', 0.1, 0, '+ 00:00:00'.$ms1, '0s'),
```
Comment (maidmaid) [inline src/Symfony/Component/VarDumper/Tests/Caster/DateCasterTest.php]: It's here the added test.

**Comment by nicolas-grekas** [comment]
can you please merge that PR with #24030?
please group PRs as much as possible, each PRs adds its own process overhead...

**Comment by maidmaid** [comment]
That's 2 different things to me: this PR is a bug in interval caster and #24030 is a feature in period caster.

**Comment by nicolas-grekas** [comment]
but both conflict, isn't it? so they do relate :)
and since this is 3.4 only, we don't really care having a dedicated line in the changelog either.

**Comment by maidmaid** [comment]
Ok, I can understand that it makes your life easier. I'll merge both PRs asap :)

**Comment by nicolas-grekas** [comment]
thanks on behalf of the core team ;)

**Comment by maidmaid** [comment]
As requested, PR merged in #24030.

**Pull request closed without merging**